test/Makefile.am

```diff
@@ -1,3 +1,3 @@
-SUBDIRS = gie gigs
+SUBDIRS = gie gigs cpp
```

Comment:
`test/cpp` - this is an unfortunate name as too generic, assuming the tests may be re-organized at some point, as @kbevers suggested on the list.

The structure could aim for something like 
```
test/gigs
test/gie
test/unit
test/functional - might or might not be needed, as per the recent discussion on list
test/apps
...
```

Reply:
OK see 6fbfa8eead3dca8edd427bbf4abf496cc3366d14

Reply:
I wished @kbevers commented on my comment first in case he disagrees, but LGMT, thanks!

Reply:
@mloskot I think your suggested structure looks good